### ISSUE TITLE
Add ability to add headers to file and directory responses

### ISSUE THREAD

**Issue by abritinthebay** [hapijs/inert#83]
Would be fantastic if - for various reasons - file and directory handlers could also set headers. This is especially useful for setting simple surrogate keys for CDNs (eg, fastly) but useful in other areas too.

I would imagine it would just be a simple key:value object passed in the options.

Thoughts?

**Comment by kanongil** [comment]
I agree that custom headers will expand the possible uses for the `directory` and `file` handler, given that it is not possible to install route specific post handler hooks.

Something like an `additionalHeaders` option with an object containing key-value could probably work.

**Comment by abritinthebay** [comment]
Yeah, that was my thought. I like the idea of naming it _additional_ headers - makes it clear that it's not a complete list.

**Comment by ErickWendel** [comment]
Guys, I implemented the first feature, Add ability to add headers to file responses, and I'm working to add headers to directory too.

**Comment by kanongil** [comment]
Hmm, I might have to revise my stance on this feature, as my initial assertion is incorrect, and it causes confusion when applied to directory handlers (eg. should it apply to a directory listing?).

The proper way to do this, is to use an ext handler:

```js
server.route({
    method: 'GET',
    path: '/{param*}',
    handler: {
        directory: {
            path: 'public'
        }
    },
    config: {
        ext: {
            onPostHandler(request, h) {

                if (request.response.variety === 'file') {
                    request.response.header('my-header', 'yes');
                }

                return h.continue;
            }
        }
    }
});
```

**Comment by ErickWendel** [comment]
Feature aborted, closing PR 😄
